solidity/contracts/UpdateStorageMirrorGuard.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity =0.8.19;
+
+import {Guard} from 'safe-contracts/base/GuardManager.sol';
+import {Enum} from 'safe-contracts/common/Enum.sol';
+import {IGuardCallbackModule} from 'interfaces/IGuardCallbackModule.sol';
+
+/**
+ * @title UpdateStorageMirrorGuard
+ * @notice This guard is responsible for calling the GuardCallbackModule when a change in settings of a safe is executed.
+ */
+contract UpdateStorageMirrorGuard is Guard {
+  /**
+   * @notice The address of the guard callback module
+   */
+  IGuardCallbackModule public immutable GUARD_CALLBACK_MODULE;
+
+  /**
+   * @notice A boolean that returns true if a tx is changing the safe's settings
+   */
+  bool public didSettingsChange;
+
+  /**
+   * @notice The hash of the new settings
+   */
+  bytes32 public settingsHash;
+
+  constructor(IGuardCallbackModule _guardCallbackModule) {
+    GUARD_CALLBACK_MODULE = _guardCallbackModule;
+  }
+
+  /**
+   * @notice Guard hook that is called before a Safe transaction is executed
+   */
+  // solhint-disable no-unused-vars
+  function checkTransaction(
+    address _to,
+    uint256 _value,
+    bytes memory _data,
+    Enum.Operation _operation,
+    uint256 _safeTxGas,
+    uint256 _baseGas,
+    uint256 _gasPrice,
+    address _gasToken,
+    address payable _refundReceiver,
+    bytes memory _signatures,
+    address _msgSender
+  ) external {
+    didSettingsChange = true;
+    settingsHash = keccak256(abi.encodePacked('settings'));
```

Comment:
At what point do we compute the plain text of the settings hash, if its here and you are leaving that part for later can you add a note